packages/react-instantsearch-core-next/src/useInstantSearch.ts

```diff
@@ -0,0 +1,31 @@
+import instantsearch from 'instantsearch.js';
+import { useEffect, useMemo, version as ReactVersion } from 'react';
+
+import version from './version';
+
+import type { InstantSearchOptions } from 'instantsearch.js';
+
+export type UseInstantSearchProps = InstantSearchOptions;
+
+export function useInstantSearch(props: UseInstantSearchProps) {
+  const search = useMemo(() => instantsearch(props), [props]);
+
+  useEffect(() => {
+    if (typeof props.searchClient.addAlgoliaAgent === 'function') {
+      props.searchClient.addAlgoliaAgent(`react (${ReactVersion})`);
+      props.searchClient.addAlgoliaAgent(
+        `react-instantsearch-core (${version})`
+      );
+    }
+  }, [props.searchClient]);
+
+  useEffect(() => {
+    search.start();
```

Comment:
future: we could have a prop/way to *not* start automatically, but only once people request it, so that search interfaces that don't display anything on empty query don't have to use the conditional requests guide

Reply:
So basically the user would have to manually start on the first input (either from a route or an input) but do it conditionally (based on `search.started`) right?

How would you see it exposed? In lifecycles?

Reply:
It can definitely be done at a later point, but I saw it as "manual instantsearch" (component or hook) without the effect

Reply:
Yes, I can definitely see an exposed `useInstantSearch` hook (probably with a different name) used along with a `ControlledInstantSearch` widget.